universal-application-tool-0.0.1/app/views/BaseHtmlLayout.java

```diff
@@ -0,0 +1,55 @@
+package views;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+import static j2html.TagCreator.document;
+
+import j2html.tags.ContainerTag;
+import j2html.tags.DomContent;
+import j2html.tags.Tag;
+import javax.inject.Inject;
+import play.twirl.api.Content;
+
+/**
+ * Base class for all layout classes.
+ *
+ * <p>A layout class should describe the DOM contents of the head, header, nav, and footer. It
+ * should have a `render` method that takes the DOM contents for the main tag.
+ */
+public class BaseHtmlLayout {
```

Comment:
Suggestion (non-blocking): maybe this should extend `BaseHtmlView` since layouts will be responsible for rendering things like the header and footer?

Reply:
I'm still thinking on this... If we do that, should we still have all the `View` classes extend `BaseHtmlView` as well?

Reply:
Yes, I think so.

Reply:
Cool, after stewing, I think this makes sense too.